### PR TITLE
FIX: clearRegions

### PR DESCRIPTION
**Pull request by panhe-xue**
## Short description
Resolves #

## Checklist
* [ ] This PR is covered by e2e tests
* [ ] It introduces no breaking API changes

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved the `clearRegions` method to safely remove regions without modifying the original array during iteration, preventing potential issues.

- **New Features**
	- Enhanced the `destroy` method to ensure all regions are cleared when the plugin is destroyed.

### CODE REVIEW

File: src/plugins/regions.ts [hunk 1]
```diff
@@ -676,7 +676,9 @@ class RegionsPlugin extends BasePlugin<RegionsPluginEvents, RegionsPluginOptions
 
   /** Remove all regions */
   public clearRegions() {
-    this.regions.forEach((region) => region.remove())
+    const regions: Region[] = JSON.parse(JSON.stringify(this.regions))
+    regions.forEach((region) => region.remove())
+    this.regions = []
```
[the remark this answers is not in the document]
Comment (katspaugh) [reply src/plugins/regions.ts:681]: I would suggest

    const regions = this.regions.slice()

[the remark this answers is not in the document]
Comment (panhe-xue) [reply src/plugins/regions.ts]: Yes, it has been modified and I hope it can contribute to the project

[the remark this answers is not in the document]
Comment (panhe-xue) [reply src/plugins/regions.ts]: Yes, it has been modified. I hope it will be helpful to the project

[the remark this answers is not in the document]
Comment (katspaugh) [reply src/plugins/regions.ts]: Thank you, looks good!

**Review by katspaugh — APPROVED** [review]

**Pull request merged**